pre_commit/repository.py

```diff
@@ -43,7 +43,8 @@ def _read_state(prefix, venv):
     if not os.path.exists(filename):
         return None
     else:
-        return json.loads(io.open(filename).read())
+        with io.open(filename) as fh:
```

Comment:
very very minor: the rest of the code conventionally uses `f` as a variable name here -- mind changing them all to use that?

Reply:
Of course!